propagators/aws/xray/propagator_test.go

```diff
@@ -6,90 +6,137 @@ package xray
-	parentSpanID               = trace.SpanID{0x53, 0x99, 0x5c, 0x3f, 0x42, 0xcd, 0x8a, 0xd8}
-	zeroSpanIDStr              = "0000000000000000"
-	wrongVersionTraceHeaderID  = "5b00000000b000000000000000000000000"
-)
+func TestExtract(t *testing.T) {
+	type extractTestCase struct {
+		name        string
+		headerVal   string
+		wantValid   bool
+		wantTraceID string
+		wantSpanID  string
+		wantSampled bool
+	}
 
-func TestAwsXrayExtract(t *testing.T) {
-	testData := []struct {
-		traceID      string
-		parentSpanID string
-		samplingFlag string
-		expected     trace.SpanContextConfig
-		err          error
-	}{
+	tests := []extractTestCase{
+		{
+			name:        "Valid header - sampled",
+			headerVal:   "Root=1-abcdef12-1234567890abcdef12345678;Parent=1234567890abcdef;Sampled=1",
+			wantValid:   true,
+			wantTraceID: "abcdef121234567890abcdef12345678",
+			wantSpanID:  "1234567890abcdef",
+			wantSampled: true,
+		},
+		{
+			name:        "Valid header - not sampled",
+			headerVal:   "Root=1-abcdef12-1234567890abcdef12345678;Parent=1234567890abcdef;Sampled=0",
+			wantValid:   true,
+			wantTraceID: "abcdef121234567890abcdef12345678",
+			wantSpanID:  "1234567890abcdef",
+			wantSampled: false,
+		},
+		{
+			name:      "Empty header - no trace info",
+			headerVal: "",
+			wantValid: false,
+		},
 		{
-			xrayTraceID, parentID64Str, notSampled,
-			trace.SpanContextConfig{
-				TraceID:    traceID,
-				SpanID:     parentSpanID,
-				TraceFlags: traceFlagNone,
-			},
-			nil,
+			name:      "Malformed TraceID - too short",
+			headerVal: "Root=1-abc-123;Parent=1234567890abcdef;Sampled=1",
+			wantValid: false,
 		},
 		{
-			xrayTraceID, parentID64Str, isSampled,
-			trace.SpanContextConfig{
-				TraceID:    traceID,
-				SpanID:     parentSpanID,
-				TraceFlags: traceFlagSampled,
-			},
-			nil,
+			name:      "Malformed TraceID - missing delimiters",
+			headerVal: "Root=1abcdef121234567890abcdef12345678;Parent=1234567890abcdef;Sampled=1",
+			wantValid: false,
 		},
 		{
-			xrayTraceID, zeroSpanIDStr, isSampled,
-			trace.SpanContextConfig{},
-			errInvalidSpanIDLength,
+			name:      "Invalid TraceID version",
+			headerVal: "Root=2-abcdef12-1234567890abcdef12345678;Parent=1234567890abcdef;Sampled=1",
+			wantValid: false,
 		},
 		{
-			xrayTraceIDIncorrectLength, parentID64Str, isSampled,
-			trace.SpanContextConfig{},
-			errLengthTraceIDHeader,
+			name:      "Invalid SpanID format",
+			headerVal: "Root=1-abcdef12-1234567890abcdef12345678;Parent=bad-spanid;Sampled=1",
+			wantValid: false,
 		},
 		{
-			wrongVersionTraceHeaderID, parentID64Str, isSampled,
-			trace.SpanContextConfig{},
-			errInvalidTraceIDVersion,
+			name:        "Missing Sampled",
+			headerVal:   "Root=1-abcdef12-1234567890abcdef12345678;Parent=1234567890abcdef",
+			wantValid:   true,
+			wantTraceID: "abcdef121234567890abcdef12345678",
+			wantSpanID:  "1234567890abcdef",
+			wantSampled: false,
+		},
+		// {
+		// 	name:        "Invalid Sampled value",
```

Comment:
Why is this commented?

Reply:
Thats the point i'm making, it seems. It seems the logic of the `parseTraceFlag` in propagators/aws/xray/propagator.go is buggy

Reply:
The current logic implies that if the value is exactly 1 character and not "1" → return None, else → return Sampled.

So in the case of Sampled=maybe, len("maybe") == 5 → not equal to 1, so it goes to the else block, and incorrectly returns trace.FlagsSampled, meaning it treats any non-"1" string of length ≠ 1 as sampled.

Is this how it should be?

I assumed that only "1" should return → Sampled, while anything else (including "0", "maybe", "foo", etc.) → Not Sampled

If I'm mistaken, I'll adjust my tests accordingly

Reply:
That makes sense, and does need to be fixed, but in another PR.
Having a commented out test case is quite confusing though. So how about removing it, and opening an issue to fix this after this PR is merged?

Reply:
Alright, that is better. I would remove the test and add it back in the PR for the bug

Reply:
Can you open the issue as well?

Reply:
https://github.com/open-telemetry/opentelemetry-go-contrib/issues/7262